src/Cli/dotnet/commands/Package/list/ListPackageReferencesCommand.cs

```diff
@@ -30,7 +32,59 @@ private static string GetAbsolutePath(string currentDirectory, string relativePa
 
     public override int Execute()
     {
-        return NuGetCommand.Run(TransformArgs());
+        string projectFile = GetProjectOrSolution();
+        bool noRestore = _parseResult.HasOption(PackageListCommandParser.NoRestore);
+        int restoreExitCode = 0;
+
+        if (!noRestore)
+        {
+            ReportOutputFormat formatOption = _parseResult.GetValue((CliOption<ReportOutputFormat>)PackageListCommandParser.FormatOption);
+            restoreExitCode = RunRestore(projectFile, formatOption);
+        }
+
+        return restoreExitCode == 0
+            ? NuGetCommand.Run(TransformArgs(projectFile))
+            : restoreExitCode;
+    }
+
+    private int RunRestore(string projectOrSolution, ReportOutputFormat formatOption)
+    {
+        MSBuildForwardingApp restoringCommand = new MSBuildForwardingApp(argsToForward: ["-target:restore", projectOrSolution, "-noConsoleLogger"]);
+
+        int exitCode = 0;
+
+        try
+        {
+            exitCode = restoringCommand.Execute();
+        }
+        catch (Exception)
+        {
+            exitCode = 1;
+        }
+
+        if (exitCode != 0)
+        {
+            if (formatOption == ReportOutputFormat.json)
+            {
+                string jsonError =
+    "{\r\n" +
+    "   \"version\": 1,\r\n" +
+    "   \"problems\": [\r\n" +
+    "      {\r\n" +
+    $"         \"text\": \"{String.Format(CultureInfo.CurrentCulture, LocalizableStrings.Error_restore)}\",\r\n" +
+    "         \"level\": \"error\"\r\n" +
+    "      }\r\n" +
+    "   ]\r\n" +
```

Comment:
Line endings are different on Windows and Linux (and in the past Mac too, but I can't remember if Mac is the same as Linux now, or still different to both).  Plus on Windows I'm not sure if `Console.WriteLine` (well, technically `StringWriter`) detects `\n` and converts it to `\r\n`, so maybe there's risk that explicitly putting in `\r\n` will cause the output to contain `\r\r\n`.

Anyway, I'd suggest using [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string) and assume that the compiler is going to do the right thing